src/mongocxx/database.hpp

```diff
@@ -247,7 +247,9 @@ class MONGOCXX_API database {
     MONGOCXX_PRIVATE database(const class client& client, stdx::string_view name);
 
     class MONGOCXX_PRIVATE impl;
-    std::unique_ptr<impl> _impl;
+    MONGOCXX_PRIVATE std::unique_ptr<impl> _impl;
+    MONGOCXX_PRIVATE const impl& _get_impl() const;
+    MONGOCXX_PRIVATE impl* _get_impl();
```

Comment:
Wait, hold that LGTM: This still needs to have the same return type as the const overload.